File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C/s580342216_commented.c
#include <stdio.h>

int main() {
    int N, K;
    while (scanf("%d %d", &N, &K) == 2) {
        int len = 0;
        while (N > 0) {
            N /= K;
            len++;
        }
        printf("%d\n", len);
    }
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/Go/s580342216_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		line := scanner.Text()
		fields := strings.Fields(line)
		N, _ := strconv.Atoi(fields[0])
		K, _ := strconv.Atoi(fields[1])
		len := 0
		n := N
		for n > 0 {
			n /= K
			len++
		}
		fmt.Println(len)
	}
	if err := scanner.Err(); err != nil {
		fmt.Fprintln(os.Stderr, "reading input:", err)
	}
}
// End of Code
